packages/dds/merge-tree/src/test/client.applyStashedOpFarm.spec.ts

```diff
@@ -0,0 +1,192 @@
+        const [message] = reconnectMsgs.shift();
+        message.sequenceNumber = ++seq;
+        clients.forEach((c) => c.applyMsg(message));
+    }
+
+    // resubmit regenerated stashed ops
+    const reRegeneratedStashedMessages = [];
+    for (const stashedOp of stashedOps) {
+        const newMsg = stashClients[1].makeOpMessage(
+            stashClients[1].regeneratePendingOp(
+                stashedOp[0],
+                stashedOp[1],
+            ));
+        reRegeneratedStashedMessages.push(newMsg);
+    }
+
+    for (const reRegeneratedStashedOp of reRegeneratedStashedMessages) {
+        reRegeneratedStashedOp.sequenceNumber = ++stashedOpSeq;
+        stashClients.forEach((c) => c.applyMsg(reRegeneratedStashedOp));
+    }
+
+    // all clients should now be the same
+    new TestClientLogger([...clients, ...stashClients]).validate();
+
+    return seq;
+}
+
+export const defaultOptions: IMergeTreeOperationRunnerConfig & { minLength: number, clients: IConfigRange } = {
+    minLength: 16,
+    clients: { min: 2, max: 8 },
+    opsPerRoundRange: { min: 40, max: 120 },
+    rounds: 3,
+    operations: [annotateRange, removeRange],
+    growthFunc: (input: number) => input * 2,
+};
+
+describe("MergeTree.Client", () => {
+    const opts = defaultOptions;
+
+    // Generate a list of single character client names, support up to 69 clients
+    const clientNames = generateClientNames();
+
+    doOverRange(opts.clients, opts.growthFunc.bind(opts), (clientCount) => {
+        it(`applyStashedOpFarm_${clientCount}`, async () => {
+            const mt = random.engines.mt19937();
+            mt.seedWithArray([0xDEADBEEF, 0xFEEDBED, clientCount]);
+
+            const clients: TestClient[] = [new TestClient()];
+            clients.forEach(
+                (c, i) => c.startOrUpdateCollaboration(clientNames[i]));
+            stashClients = [new TestClient()];
+            stashClients.forEach(
+                (c, i) => c.startOrUpdateCollaboration(clientNames[i]));
+
+            let seq = 0;
+            clients.forEach((c) => c.updateMinSeq(seq));
+            stashClients.forEach((c) => c.updateMinSeq(seq));
+
+            // Add double the number of clients each iteration
+            const targetClients = Math.max(opts.clients.min, clientCount);
+            for (let cc = clients.length; cc < targetClients; cc++) {
+                const newClient = await TestClient.createFromClientSnapshot(clients[0], clientNames[cc]);
+                clients.push(newClient);
+                // add 1 stash client per normal client
+                const anotherNewClient = await TestClient.createFromClientSnapshot(clients[0], clientNames[cc]);
+                stashClients.push(anotherNewClient);
+            }
+
+            seq = runMergeTreeOperationRunner(
+                mt,
+                seq,
+                clients,
+                opts.minLength,
+                opts,
+                applyMessagesWithReconnect);
```

Comment:
you can remove the global of stashClients by using a function factory here
```suggestion
                (seq, md, c, l )=>applyMessagesWithReconnect(seq, md, c, l, stashClients);
```